### ISSUE TITLE
Create class for Rules, Severity & Violations

### ISSUE THREAD

**Issue by s3n-w6i** [precice-forschungsprojekt/config-checker#2]
**Short description**

To represent rules in a config file, we want to make them a class called "Rule". It has a check function that can be executed on the built graph and on the original xml-tree. This function then returns a list of Violations. The actual check will be implemented in the specific implementation of a rule (not a part of this task).
Rules have different Severities.

**Pseudocode**
```python
import enum

# These next lines are only for illustration of how we want to use them later (not in scope of this issue)
RULES = [RuleA, RuleB]

for rule in RULES:
    violations.add_all(rule.check(graph))

for violation in violations:
    print(violation)

# Implement the following:

enum Severity:
    WARN = "warning"


class Rule:
    severity = Severity("warning")

    def check(graph) -> list[Violation]


class Violation<Rule>:
    affected_nodes
```

**Depends on**
Depends on #1 to be completed
